src/blueprints/tv2/factories/tv2-gfx-action-factory.ts

```diff
@@ -0,0 +1,176 @@
+import { Tv2BlueprintConfiguration } from '../value-objects/tv2-blueprint-configuration'
+import { Action, PieceAction } from '../../../model/entities/action'
+import { PieceInterface } from '../../../model/entities/piece'
+import { PieceType } from '../../../model/enums/piece-type'
+import { Tv2SourceLayer } from '../value-objects/tv2-layers'
+import { TransitionType } from '../../../model/enums/transition-type'
+import { Tv2GfxTimelineObjectFactory } from '../value-objects/factories/tv2-gfx-timeline-object-factory'
+import { PieceLifespan } from '../../../model/enums/piece-lifespan'
+import { PieceActionType } from '../../../model/enums/action-type'
+import { Tv2DownstreamKeyer } from '../value-objects/tv2-studio-blueprint-configuration'
+
+export class Tv2GfxActionFactory {
+  constructor(
+    private readonly gfxTimelineObjectFactory: Tv2GfxTimelineObjectFactory
+  ) { }
+
+  public createGfxActions(blueprintConfiguration: Tv2BlueprintConfiguration): Action[] {
+    return [
+      this.createThemeOutAction(blueprintConfiguration),
+      this.createOverlayInitializeAction(),
+      this.createContinueGfxAction(),
+      this.createGfxClearAction(blueprintConfiguration),
+      this.createGfxAlternativeOutAction(blueprintConfiguration),
+      ...this.createDownstreamKeyerOnActions(blueprintConfiguration),
+      ...this.createDownstreamKeyerOffActions(blueprintConfiguration)
+    ]
+  }
+
+  private createDownstreamKeyerOffActions(blueprintConfiguration: Tv2BlueprintConfiguration): PieceAction[] {
+    return blueprintConfiguration.studio.SwitcherSource.DSK.map(config => this.createDownStreamKeyerAction(config, 'Off', false))
+  }
+
+  private createDownStreamKeyerAction(downstreamKeyer: Tv2DownstreamKeyer, actionName: string, isOn: boolean): PieceAction {
```

Comment:
Can't the `actionName` be inferred from `isOn`?

Reply:
Yes it can. Doing so means that instead of the boolean being a pass through argument, it changes what the method does, meaning it is a flag. 

I had created it originally like suggested, and then asked for Lindveds input, after which we reworked it to this.